File: C# Fundamentals/BasicSyntax/Vacation.cs
using System;

namespace Vacation
{
    class Program
    {
        static void Main(string[] args)
        {
            int group = int.Parse(Console.ReadLine());
            string type = Console.ReadLine().ToLower();
            string day = Console.ReadLine().ToLower();

            double price = 0;

            switch (type)
            {
                case "students":
                    switch (day)
                    {
                        case "friday":
                            price = 8.45;
                            break;

                        case "saturday":
                            price = 9.80;
                            break;

                        case "sunday":
                            price = 10.46;
                            break;
                    }
                    break;

                case "business":

                    if (group >= 100)
                    {
                        group -= 10;
                    }

                    switch (day)
                    {
                        case "friday":
                            price = 10.90;
                            break;

                        case "saturday":
                            price = 15.60;
                            break;

                        case "sunday":
                            price = 16;
                            break;
                    }
                    break;

                case "regular":
                    switch (day)
                    {
                        case "friday":
                            price = 15;
                            break;

                        case "saturday":
                            price = 20;
                            break;

                        case "sunday":
                            price = 22.5;
                            break;
                    }
                    break;
            }

            double totalPrice = group * price;
            if (type == "students" && group >= 30)
            {
                totalPrice *=  (1 - 15 / 100.0);
            }
            if (type == "regular" && group >= 10 && group <= 20)
            {
                totalPrice *= (1 - 5 / 100.0);
            }

            Console.WriteLine($"Total price: {totalPrice:f2}");
        }
    }
}

File: Java Fundamentals/BasicSyntaxEx/Vacation.java
package BasicSyntaxEx;

import java.util.Scanner;

public class Vacation {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        int groupNumber = Integer.parseInt(scanner.nextLine());
        String type = scanner.nextLine().toLowerCase();
        String day = scanner.nextLine();

        double price = 0d;
        switch (day) {
            case "Friday":
                switch (type) {
                    case "students":
                        price = 8.45;
                        break;
                    case "business":
                        price = 10.90;
                        break;
                    case "regular":
                        price = 15;
                        break;
                }
                break;
            case "Saturday":
                switch (type) {
                    case "students":
                        price = 9.80;
                        break;
                    case "business":
                        price = 15.60;
                        break;
                    case "regular":
                        price = 20;
                        break;
                }
                break;
            case "Sunday":
                switch (type) {
                    case "students":
                        price = 10.46;
                        break;
                    case "business":
                        price = 16;
                        break;
                    case "regular":
                        price = 22.50;
                        break;
                }
                break;
        }

        if (type.equals("students") && groupNumber >= 30) {
            price *= (1 - (15 / 100d));
        } else if (type.equals("regular") && groupNumber >= 10 && groupNumber <= 20) {
            price *= (1 - (5 / 100d));
        } else if (type.equals("business") && groupNumber >= 100) {
            groupNumber -= 10;
        }

        System.out.printf("Total price: %.2f", price * groupNumber);
    }
}
